### DISCUSSION TITLE
Divinity Original Sin Enhanced Edition: 无法启动

### DISCUSSION THREAD

**Discussion by the0demiurge** [Gamuxorg/bbs#307] [游戏或网站运行反馈]
### 以下是推荐模板
游戏名称：
Divinity Original Sin Enhanced Edition
操作系统：*（uname -a）*
Arch Linux
bug描述：*（具体说说发生了什么）*

终端运行输出 ：*（如果有的话）*
```
ATTENTION: default value of option vblank_mode overridden by environment.
Thread "EoCApp" (3353160768)
received signal 11

Call stack:

(0) /usr/lib/libpthread.so.0 : +0x14800 [0x7f42c840e800]
(1) /tmp/.mount_dosee_Uy02sC/usr/bin/libOGLBinding.so : api::OpenGLRenderer::ApplyConstants()+0x65 [0x7f42c8d87845]
(2) /tmp/.mount_dosee_Uy02sC/usr/bin/libRenderFramework.so : rf::Renderer::Apply(bool)+0x57 [0x7f42c8a2d437]
(3) /tmp/.mount_dosee_Uy02sC/usr/bin/EoCApp : ig::IggyBinding::Swap(rf::Renderer*)+0xfc [0xed032c]
(4) /tmp/.mount_dosee_Uy02sC/usr/bin/libGameEngine.so : BaseApp::EndDrawGUI(rf::Renderer*)+0x9b [0x7f42c8b87fab]
(5) /tmp/.mount_dosee_Uy02sC/usr/bin/libGameEngine.so : BaseApp::MakeFrame()+0x3a4 [0x7f42c8b884d4]
(6) /tmp/.mount_dosee_Uy02sC/usr/bin/libGameEngine.so : BaseApp::OnIdle()+0xe0 [0x7f42c8b86cb0]
(7) /tmp/.mount_dosee_Uy02sC/usr/bin/EoCApp : main+0x170 [0x6d5180]
(8) /usr/lib/libc.so.6 : __libc_start_main+0xf3 [0x7f42c825b023]
(9) /tmp/.mount_dosee_Uy02sC/usr/bin/EoCApp : _start+0x29 [0x6d4ef9]
/tmp/.mount_dosee_Uy02sC/usr/bin//game.sh: line 4:  8699 Segmentation fault      (core dumped) LD_LIBRARY_PATH="$HERE" $HERE/EoCApp

```
自行尝试解决bug的方式和结果：*（如果有的话）*

https://steamcommunity.com/app/373420/discussions/0/451848854988674764/?l=schinese

bijineau 提供的方案好用：

> I had the same problem initially (ArchLinux). I think AftershockVibe was on the right path. The system requirements says OpenGL 4.0 so it may be that your video card/driver doesn't support it. For example intel/mesa only supports 3.0.
> 
> This command should tell you the version:
> glxinfo | grep "OpenGL version"
> 
> OPTIMUS USERS
> the nvidia binary driver is OpenGL 4.5 . I'm running on an optimus laptop, so I have to launch through primusrun. I was still getting the crash. Turns out, Divinity uses a wrapper shell-script which screws up primusrun causing the actual application binary to run without primus/optimus.
> 
> To fix this temporarily, I edited
> <path to library>/SteamApps/common/Divinity Original Sin Enhanced Edition/runner.sh
> to have it use primusrun:
> LD_LIBRARY_PATH="." primusrun ./EoCApp
> . It launched successfully after that (using the nvidia card).

在`game.sh`中加入primusrun。建议魔改一下game.sh，判断有primusrun或optirun的话就自动加入，不然报一个warning

**Comment by shouhuanxiaoji** [comment]
收到，可能是nvidia显卡驱动问题，开源驱动有可能无法运行。我们确认一下
